HW4/HW4.py

```diff
@@ -0,0 +1,381 @@
+    # Determine if response is continuous or boolean
+    y = response
+    if df[response].unique().size > 2:
+        response_type = "continuous"
+    else:
+        # hard coding to boolean rather than categorical
+        response_type = "boolean"
+
+    # Determine if predictors are categorical or continuous
+    cont_master_list = df._get_numeric_data().columns.to_list()
+    predictor_types = {}
+    cont_predictor_col_names = []
+    for predictor in predictors:
+        predictor_type = ""
+        if predictor in cont_master_list:
+            predictor_type = "continuous"
+        if True in df[predictor].unique().tolist():
+            predictor_type = "boolean"
+        if predictor_type == "":
+            if df.dtypes[predictor] == "O":
+                predictor_type = "categorical"
+        if predictor_type == "":
+            predictor_type = input(
+                f"Toss me a bone, what's the predictor type for {predictor}? "
+                f"(continuous, categorical, or boolean):"
+            )
+        if predictor_type == "":
+            print("that's not very nice")
+            predictor_type = "categorical"
+        predictor_types[predictor] = predictor_type
+        if predictor_type == "continuous":
+            cont_predictor_col_names.append(predictor)
+
+    # Heatplot
+    def Heatplot(x):
+        conf_matrix = confusion_matrix(df[x].astype(str), df[y].astype(str))
+        fig_cat_cat = go.Figure(
+            data=go.Heatmap(z=conf_matrix, zmin=0, zmax=conf_matrix.max())
+        )
+        fig_cat_cat.update_layout(
+            title="",
+            xaxis_title="Response",
+            yaxis_title="Predictor",
+        )
+        filename = x + " heatmap.html"
+        fig_cat_cat.write_html(file=filename, include_plotlyjs="cdn")
+        return filename
+
+    def Violin(x):
+        # violin plot on predictor grouped by response
+        fig_2 = go.Figure(
+            data=go.Violin(
+                y=df[y],
+                x=df[x],
+                line_color="black",
+                box_visible=True,
+                meanline_visible=True,
+            )
+        )
+        fig_2.update_layout(
+            title=x + " Violin",
+            xaxis_title="Response",
+            yaxis_title="Predictor",
+        )
+        filename = x + " violin.html"
+        fig_2.write_html(file=filename, include_plotlyjs="cdn")
+        return filename
+
+    def Distribution(x):
+        # distribution plot on predictor grouped by response
+        # Create distribution plot with custom bin_size
+        hist_data = [df[x]]
+        group_labels = [x]
+        fig_1 = ff.create_distplot(hist_data, group_labels)
+        fig_1.update_layout(
+            title=f"Distribution for {x}",
+            xaxis_title="Predictor",
+            yaxis_title="Distribution",
+        )
+        filename = x + "cont predict by cat resp distplot.html"
+        fig_1.write_html(file=filename, include_plotlyjs="cdn")
+
```

Comment:
Yeah, I'm not sure what I was thinking here? I think im going to give it a night's rest before refactoring anything since I'm probably just going to make it worse at this point.